mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -508,6 +678,101 @@ getSizesAndStrides(int64_t rank1, StridedMemRefType<float, 4> *filter,
               outputSizeStride['h'].second, outputSizeStride['w'].second};
   return;
 }
+static void
+getSizesAndStrides(int64_t rank1, StridedMemRefType<float, 5> *filter,
+                   int64_t rank2, StridedMemRefType<float, 5> *input,
+                   int64_t rank3, StridedMemRefType<float, 5> *output,
+                   void *f_layout, void *i_layout, void *o_layout,
+                   llvm::SmallVector<int64_t, 5> &fSizes,
+                   llvm::SmallVector<int64_t, 5> &fStrides,
+                   llvm::SmallVector<int64_t, 5> &iSizes,
+                   llvm::SmallVector<int64_t, 5> &iStrides,
+                   llvm::SmallVector<int64_t, 5> &oSizes,
+                   llvm::SmallVector<int64_t, 5> &oStrides) {
+  auto filterSizes = llvm::ArrayRef<int64_t>(filter->sizes, rank1);
+  auto filterStrides = llvm::ArrayRef<int64_t>(filter->strides, rank1);
+
+  auto inputSizes = llvm::ArrayRef<int64_t>(input->sizes, rank2);
+  auto inputStrides = llvm::ArrayRef<int64_t>(input->strides, rank2);
+
+  auto outputSizes = llvm::ArrayRef<int64_t>(output->sizes, rank3);
+  auto outputStrides = llvm::ArrayRef<int64_t>(output->strides, rank3);
+
+  auto *layout1 = static_cast<StridedMemRefType<char, 1> *>(f_layout);
+  auto *filterLayout = layout1->data + layout1->offset;
+
+  auto *layout2 = static_cast<StridedMemRefType<char, 1> *>(i_layout);
+  auto *inputLayout = layout2->data + layout2->offset;
+
+  auto *layout3 = static_cast<StridedMemRefType<char, 1> *>(o_layout);
+  auto *outputLayout = layout3->data + layout3->offset;
+
+  // Extract tensor sizes and strides into a map
+  std::unordered_map<char, std::pair<int64_t, int64_t>> filterSizeStride,
+      inputSizeStride, outputSizeStride;
+  for (size_t i = 0; i < 5; i++) {
+    filterSizeStride[filterLayout[i]] =
+        std::make_pair(filterSizes[i], filterStrides[i]);
+    inputSizeStride[inputLayout[i]] =
+        std::make_pair(inputSizes[i], inputStrides[i]);
+    outputSizeStride[outputLayout[i]] =
+        std::make_pair(outputSizes[i], outputStrides[i]);
+  }
+
+  // Move sizes and strides into vectors to help compiler optimization
+  // filter: k c y x
```

Comment:
k c y x -> g k c y x?

other comments in the function also need to be revised.